core/service.go

```diff
@@ -38,23 +38,28 @@ var _ services.Service = &Service{}
 // BABE session, and p2p service. It deals with the validation of transactions
 // and blocks by calling their respective validation functions in the runtime.
 type Service struct {
-	rt      *runtime.Runtime
-	bs      *babe.Session
-	blkRec  <-chan types.Block // receive blocks from BABE session
-	msgRec  <-chan p2p.Message // receive messages from p2p service
-	msgSend chan<- p2p.Message // send messages to p2p service
+	blockState   BlockState
+	storageState StorageState
+	rt           *runtime.Runtime
+	bs           *babe.Session
+	blkRec       <-chan types.Block // receive blocks from BABE session
+	msgRec       <-chan p2p.Message // receive messages from p2p service
+	msgSend      chan<- p2p.Message // send messages to p2p service
 }
 
 type Config struct {
-	Keystore *keystore.Keystore
-	Runtime  *runtime.Runtime
-	MsgRec   <-chan p2p.Message
-	MsgSend  chan<- p2p.Message
+	BlockState   BlockState
+	StorageState StorageState
+	Keystore     *keystore.Keystore
+	Runtime      *runtime.Runtime
+	MsgRec       <-chan p2p.Message
+	MsgSend      chan<- p2p.Message
+	NewBlocks    chan types.Block // only used for testing purposes
```

Comment:
Why do we need this?

Reply:
there's a test that needs to pass in the NewBlocks channel so that it can pass in a message (core TestAnnounceBlock)